src/transformers/models/falcon/modeling_falcon.py

```diff
@@ -129,6 +129,12 @@ def cos_sin(
         total_length = seq_len + past_key_values_length
         if total_length > self.seq_len_cached:
             self._set_cos_sin_cache(total_length, device, dtype)
+
+        # the cached tensors need to update their devices (for example, after we change the model's device)
+        if device != self.cos_cached:
+            self.cos_cached = self.cos_cached.to(device)
+            self.sin_cached = self.sin_cached.to(device)
```

Comment:
without this, we have problems when switching a model device

Reply:
Yes, don't know why I made such mistake (as it works on the code snippet I am using 😅  sorry). You saved me!